deploy/charts/operator-crds/crd-helm-wrapper/main.go

```diff
@@ -0,0 +1,335 @@
+// Copyright 2025 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// crd-helm-wrapper is a tool that wraps Kubernetes CRD YAML files with Helm
+// template conditionals for conditional installation, feature flags,
+// and resource policy annotations.
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"embed"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"gopkg.in/yaml.v3"
+)
+
+//go:embed templates/*.tpl
+var templateFS embed.FS
+
+// Templates holds the loaded template content
+type Templates struct {
+	Header         string
+	Footer         string
+	HeaderFeature  string
+	FooterFeature  string
+	KeepAnnotation string
+}
+
+// CRDMetadata represents the minimal structure needed to extract CRD name
+type CRDMetadata struct {
+	Kind     string `yaml:"kind"`
+	Metadata struct {
+		Name string `yaml:"name"`
+	} `yaml:"metadata"`
+}
+
+// Config holds the wrapper configuration
+type Config struct {
+	SourceDir string
+	TargetDir string
+	Verbose   bool
+}
+
+// CRD feature flag groups
+// Maps CRD name prefixes to their corresponding Helm values flag
+var crdFeatureFlags = map[string]string{
+	// Server group: mcpservers, mcpexternalauthconfigs, mcpremoteproxies, mcptoolconfigs, mcpgroups
+	"mcpservers":             "enableServer",
+	"mcpexternalauthconfigs": "enableServer",
+	"mcpremoteproxies":       "enableServer",
+	"mcptoolconfigs":         "enableServer",
+	"mcpgroups":              "enableServer",
+	// Registry group: mcpregistries
+	"mcpregistries": "enableRegistry",
+	// VirtualMCP group: virtualmcpservers, virtualmcpcompositetooldefinitions
+	"virtualmcpservers":                  "enableVirtualMcp",
+	"virtualmcpcompositetooldefinitions": "enableVirtualMcp",
+}
+
+func main() {
+	cfg := Config{}
+
+	flag.StringVar(&cfg.SourceDir, "source", "", "Source directory containing raw CRD YAML files")
+	flag.StringVar(&cfg.TargetDir, "target", "", "Target directory for wrapped Helm templates")
+	flag.BoolVar(&cfg.Verbose, "verbose", false, "Enable verbose output")
+	flag.Parse()
+
+	if cfg.SourceDir == "" || cfg.TargetDir == "" {
+		fmt.Fprintln(os.Stderr, "Usage: crd-helm-wrapper -source <dir> -target <dir>")
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	if err := run(cfg); err != nil {
+		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func loadTemplates() (*Templates, error) {
+	header, err := templateFS.ReadFile("templates/header.tpl")
```

Comment:
would it make any sense to embed the files via go:embed? I guess it doesn't make much difference since we don't ship the result right?